core/primitives/address.hpp

```diff
@@ -3,16 +3,146 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#ifndef CMAKE_HUNTER_SEED_CORE_ADDRESS_ADDRESS_HPP
-#define CMAKE_HUNTER_SEED_CORE_ADDRESS_ADDRESS_HPP
+#ifndef CPP_FILECOIN_CORE_ADDRESS_HPP
+#define CPP_FILECOIN_CORE_ADDRESS_HPP
+
+#include <cstdint>
+#include <memory>
+#include <string>
+#include <type_traits>
+#include <vector>
+
+#include "boost/variant.hpp"
+#include "common/outcome.hpp"
 
 namespace fc::primitives {
+  using Uint = uint64_t;            // TODO:
+  using Byte = uint8_t;             // consider moving
+  using Bytes = std::vector<Byte>;  // commonly used types
```

Comment:
Fair enough. It's just what they are called in the spec so arguably it could be easier to match things up. But it didn't seem right still, hence the "TODO" comment